test/new-e2e/utils/e2e/client/stack_initializer.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+)
+
+type stackInitializer interface {
+	setStack(auth *Authentification, stackResult auto.UpResult) error
+}
+
+type Authentification struct {
+	SSHKey string
+}
+
+func CheckEnvStructValid[Env any]() error {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should `Env` be more limited to interface implementing `Execute(command string)` ?

Reply:
`Env` doesn't implement anything. The fields of `Env` should implements `stackInitializer`.